src/ArcgisAccount/useAccountManager.js

```diff
@@ -0,0 +1,189 @@
+  const logoutAccount = useCallback(
+    key => {
+      const { accounts } = accountManagerState || {};
+      if (accounts[key]) {
+        const { session, token } = accounts[key];
+        const portalUrl = session ? session.portal : null;
+        const clientId = session ? session.clientId : null;
+        logoutOAuth2({
+          url: portalUrl,
+          clientId: clientId,
+          token: token
+        });
+
+        //Update localStorage/ state
+        logoutAccountStorage(name, key);
+        const accountManager = getAccountManagerStorage(name);
+        setAccountManagerState(accountManager);
+      }
+    },
+    [accountManagerState, name]
+  );
+
+  /** Remove Account: Remove account from local storage and attempt revoke token*/
+  const removeAccount = useCallback(
+    key => {
+      const { accounts } = accountManagerState || {};
+
+      if (accounts && accounts[key]) {
+        if (accounts[key].token) {
+          const { session, token } = accounts[key];
+          const portalUrl = session ? session.portal : null;
+          const clientId = session ? session.clientId : null;
+          //Remove token and session
+          logoutOAuth2({
+            url: portalUrl,
+            clientId: clientId,
+            token: token
+          });
+        }
+
+        //Update localStorage/ state
+        removeAccountStorage(name, key);
+        const accountManager = getAccountManagerStorage(name);
+        setAccountManagerState(accountManager);
+      }
+    },
+    [accountManagerState, name]
+  );
+
+  /** Restore Account: Log in to an existing account that is logged out */
+  const restoreAccount = options => {
+    const selectOptions = options ? options : authOptions;
+    const { clientId, redirectUri, portalUrl, popup } = selectOptions || {};
+
+    //set localstorage status
+    beginStatusStorage(name, options);
+    //begin login
+    loginOAuth2({ clientId, redirectUri, portalUrl, popup });
+  };
+
+  /** Refresh Account: UserSession.refreshSession [https://esri.github.io/arcgis-rest-js/api/auth/UserSession/#refreshSession] */
+  const refreshAccount = key => {
+    const { accounts } = accountManagerState || {};
+    const { session } = accounts[key];
+    session.refreshSession().then(session => {
+      const sSession = session.serialize();
+      refreshAccountStorage(name, key, sSession);
+    });
+  };
+
+  //** Set the active account */
+  const switchActiveAccount = useCallback(
+    key => {
+      switchActiveStorage(name, key);
+      const accountManager = getAccountManagerStorage(name);
+      setAccountManagerState(accountManager);
+    },
+    [name]
+  );
+
+  return {
+    accountManagerState,
+    addAccount,
+    logoutAccount,
+    removeAccount,
+    restoreAccount,
+    refreshAccount,
+    switchActiveAccount,
+    getUserThumbnail,
+    getOrgThumbnail
+  };
+};
+
+useAccountManager.propTypes = {
```

Comment:
I'm not sure prop types do anything if it's not a react component, do we want to keep this for our own documentation though?

Reply:
will remove until better solution for the table is found